src/type/validate.js

```diff
@@ -165,7 +179,10 @@ function getOperationTypeNode(
   return type.astNode;
 }
 
-function validateDirectives(context: SchemaValidationContext): void {
+function validateDirectiveDefinitions(context: SchemaValidationContext): void {
+  // Ensure no directive is defined multiple times
+  const directiveDefinitions = new Map();
+
```

Comment:
@mjmahone This change is make sense on it's own since you can manually construct `GraphQLSchema` with duplicated directives.
Can you extract this change into separate PR?

Reply:
On `Map`: all around this library, we use `Object.create(null)` for maps with strings as keys and `Map` for non-string keys. AFAIK `Map` have significantly lover performance, I'm also not sure about memory usage.